### DISCUSSION TITLE
Regex for Value Mappings in Table Panels feature

### DISCUSSION THREAD

**Discussion by yuanbit** [grafana/grafana#37008] [Dashboards, Panels & Visualizations]
I would like the value mappings in table panels to have the feature to apply regex to a value then map the regex match to be the new text. 

For example if my value is: 03.03.2019 16:19:33, error, 12345

I want to map each row with the above format to a text of only the last digits (12345) as they represent the error ids.

Is there a way to do this already? If not, could this feature be added?

**Comment by torkelo** [comment]
This is currently not supported. Not sure I understand what you mean either. 

But does not sound like common use case. 

Closing this issue, just because it is closed does not mean it will never be implemented. It can still act as a feature request and gain +1/up votes in order to see user interest in the feature. Just means it's not something the core team plans to work on in next 1+ year. We are trying to limit the number of open issues to better manage & plan work.

**Comment by DranonJoD** [comment]
This or a way to substring, especially for pod name in docker that can be long but part of the name is not useful to identify which it is (naming convention at my work makes the pod name contain the service name for example).

**Comment by keremispirli** [comment]
A way to map a value to a substring of that value would be great. Regex support would be useful for this.

**Comment by grzesuav** [comment]
I can see value when applying to elasticsearch DataSource (extract relevant part from the log and display it)

**Comment by jehutywong** [comment]
+1

**Comment by xurxofp** [comment]
This should be pretty useful

**Comment by i-takizawa** [comment]
Really hope this gets implemented.

My use case:
Since I'm deploying node_exporter in docker container, filesystem mountpoints on `/*` are labeled as `/host/*`. It would be nice if I could just replace them all with `/*` in a single statement.

**Comment by JosePMTecEng** [comment]
+1

**Comment by valenzuelaomar** [comment]
+1

**Comment by binhvcom** [comment]
+1

**Comment by mammuthus** [comment]
+1

**Comment by gnuneznetuy** [comment]
+1

**Comment by omalisson** [comment]
+1

**Comment by eperret** [comment]
+1

**Comment by eocern** [comment]
Would love to see that implemented too !

**Comment by covelloz** [comment]
+1

**Comment by zoltanbedi** [comment]
@grafana/grafana-frontend-platform should we reopen this issue?

**Comment by ox-rjaumain** [comment]
+1

Yes please, consider reopening.

**Comment by putsev** [comment]
+1

**Comment by putsev** [comment]
My usage is convert integer value of version of application to beautiful string value (for example 1000234 convert to 1.0.2.34). I know than i can do this through labels, but this is additional row in prometheus table

**Comment by RishoT17** [comment]
+1
My use case is cells containing URL links that are used by in data-links. It would be useful to display a human-readable label instead of the URL link.
Related to: https://github.com/grafana/grafana/issues/24482#issuecomment-841180937_

**Comment by alexbergamo** [comment]
+1

**Comment by torkelo** [comment]
If anyone is interesting in helping out with this it should be quite easy to do: 

Here are the value mapping types & option models:
https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/valueMapping.ts#L4

Here is the value mapping logic handled: 
https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/valueMappings.ts#L7

Here is the editor for each row:
https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingEditRow.tsx

The edit view model is a bit different so need new code here:
https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditorModal.tsx#L158

**Comment by nibors** [comment]
+1

**Comment by mcdee** [comment]
Thanks to the pointers from @torkelo I have created a PR for this function at https://github.com/grafana/grafana/pull/38931

**Reply by pkolyvas** [reply]
@mcdee We're really excited about this. Thanks so much for the contribution.

**Reply by imReker** [reply]
It's not work on numbers or boolean, only works on Strings 😞

**Reply by mcdee** [reply]
regexes are generally for string values.  Do you have any particular use cases for numbers or booleans with regexes?

**Reply by imReker** [reply]
For example:
1. Different colors for odd/even values in table.
2. Different colors for different values in Timeline States (by setting 10 colors to 0-9 single digits).

**Comment by offermannu** [comment]
+1

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This feature seems to be mainly resolved with https://github.com/grafana/grafana/pull/38931, so I will close this discussion. If there is an aspect of this feature that was not addressed, please create a new issue detailing what you'd like to see done, and it will be triaged and prioritized along with the team's other work.
